analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,415 @@
+  # the gene as that status if more than 50% of the total counts are
+  # for that particular status
+  gene_status_count <- status_majority_caller(neutral_status_cytoband_df,
+                                              "gene_symbol",
+                                              "status", 
+                                              percent_threshold = percent_threshold)
+  
+  # Display table
+  gene_status_count
+}
+```
+
+## Plot calls
+
+Plot the final dominant status call on the x-axis and the percent of each status on the y-axis.
+
+### Plot chromosome arm status calls
+
+```{r}
+# Run `plot_dominant_status_calls` function for the chromosome arm calls
+plot_dominant_status_calls(
+  arm_status_count,
+  "chromosome_arm"
+)
+```
+
+### Plot cytoband status calls
+
+```{r fig.width = 10}
+# Run `plot_dominant_status_calls` function for the cytoband calls if not
+# running in CI
+if (!(running_in_ci)) {
+  plot_dominant_status_calls(cytoband_status_count,
+                             "cytoband")
+}
+```
+
+### Plot gene status calls
+
+```{r}
+# Plot if not running in circleCI
+if (!(running_in_ci)) {
+  plot_dominant_status_calls(gene_status_count,
+                             "gene_symbol")
+}
+```
+
+## Combine arm, cytoband, and gene level status data
+
+```{r}
+# The logic variable `running_in_ci` is needed here because the CI testing
+# files do not contain any of the genes in the `gene_status_count` data.frame
+# generated above (when `running_in_ci` == FALSE)
+if (!(running_in_ci)) {
+  final_df <- consensus_seg_autosomes_df %>%
+    mutate(chromosome_arm = gsub("(p|q).*", "\\1", cytoband)) %>%
+    inner_join(arm_status_count,
+               by = "chromosome_arm") %>%
+    left_join(cytoband_status_count,
+              by = "cytoband",
+              suffix = c(".arm", ".cytoband")) %>%
+    left_join(gene_status_count,
+              by = "gene_symbol",
+              suffix = c(".arm", ".gene")) %>%
+    mutate(
+      focal_call = paste0(
+        dominant_status.arm,
+        ", ",
+        dominant_status.cytoband,
+        ", ",
+        dominant_status
+      ),
+      # Here we want to define the most focal call based on the arm, cytoband,
+      # and gene status information -- if a loss/gain is defined at the arm
+      # level then the focal call will be "arm_loss" or "arm_gain" respectively,
+      # and so on.
+      focal_call = case_when(focal_call == "loss, NA, NA" ~ "arm_loss",
+                             focal_call == "neutral, uncallable, NA" ~ "uncallable",
+                             focal_call == "neutral, loss, NA" ~ "cytoband_loss",
+                             focal_call == "neutral, NA, NA" ~ "arm_neutral",
+                             focal_call == "neutral, unstable, neutral" ~ "gene_neutral",
+                             focal_call == "neutral, unstable, loss" ~ "gene_loss",
+                             TRUE ~ "Other")
+    )
+} else {
+  final_df <- consensus_seg_autosomes_df %>%
```

Comment:
So in CI it won't do the recoding?

Reply:
Oops, I'll add the recoding here.

Reply:
I think we might refactor this section once we determine what `final_df` needs to look like, so you can hold off for now.

Reply:
Okay sounds good